src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1231,6 +1231,9 @@ export class SavedObjectsRepository {
     if (!this._allowedTypes.includes(type)) {
       throw SavedObjectsErrorHelpers.createGenericNotFoundError(type, id);
     }
+    if (!id) {
+      throw SavedObjectsErrorHelpers.createBadRequestError('id cannot be empty'); // prevent potentially upserting a saved object with an empty ID
+    }
```

Comment:
If the intent of the PR is to avoid creating objects with empty ids, should we only throw if `options.upsert` is set, both for `update` and `incrementCounter`? OTOH it doesn't make sense to try to update an object with empty id, but in that case, maybe we should ensure this for all methods?

But the PR does what it's supposed to do, protect against introducing corrupted docs, so it's fine with me.

Reply:
Yeah, I wanted to minimize the changes but at least keep the responses consistent within the `update` method. I think we should probably ensure this for all methods (along with other ID validation) but I'd prefer to leave it for the follow-on issue :+1: